tests/test_plugin.py

```diff
@@ -29,22 +31,41 @@ def test_adding_widget_to_viewer(viewer_widget):
 
 def test_basic_function(qtbot, viewer_widget):
     viewer, widget = viewer_widget
-    viewer.open_sample('cellpose-napari', 'rgb_2D.png')
+    viewer.open_sample('cellpose-napari', 'rgb_2D')
+    viewer.layers[0].data = viewer.layers[0].data[0:128, 0:128]
 
-    if os.getenv("CI"):
-        return
+    #if os.getenv("CI"):
+    #    return
         # actually running cellpose like this takes too long and always timesout on CI
         # need to figure out better strategy
-    widget.compute_diameter_button.changed(None)
-    widget()  # run segmentation
+    widget()  # run segmentation with all default parameters
 
     def check_widget():
         assert widget.cellpose_layers
 
     qtbot.waitUntil(check_widget, timeout=30_000)
+    # check that the layers were created properly
     assert len(viewer.layers) == 5
     assert "cp_masks" in viewer.layers[-1].name
 
+    # check that the segmentation was proper, should yield 11 cells
+    assert viewer.layers[-1].data.max() == 11
+
+def test_compute_diameter(qtbot, viewer_widget):
```

Comment:
still stalling... tried it a few times.  You can skip this on linux if you want

Reply:
I'll take a look later, maybe checking the value of the LineEdit isn't the right approach.